core/templates/services/search.service.ts

```diff
@@ -229,12 +233,17 @@ export class SearchService {
   }
 
   loadMoreData(
-      successCallback: (SearchResponseData, boolean) => void,
-      failureCallback?: (boolean) => void): void {
+      successCallback: (
+        SearchResponseData: SearchResponseBackendDict,
+        boolean: boolean
+      ) => void,
+      failureCallback?: (arg0: boolean) => void): void {
```

Comment:
```suggestion
      failureCallback?: (arg0: boolean) => void
  ): void {
```
And please add a comment above on why the `failureCallback` is optional.

Reply:
Done, added a comment